app/policies/market_model_policy.rb

```diff
@@ -1,6 +1,10 @@
 class MarketModelPolicy < ApplicationPolicy
   include PrivatePolicy
 
+  def destroy?
+    super && TestingGround.where(market_model: record).count <= 1
```

Comment:
Instead of `<= 1` – which would theoretically allow a market model to be deleted while one LES was still using it – could the LES be deleted *first*, and then just check if the count is zero?

I'm actually a bit on the fence about mixing the Policy/authorization stuff into the TestingGround model, I feel like that's something which belongs at the level of the controller (or a service object).

Perhaps this might be encapsulated as a `TestingGround::Destroy` service (like [`SaveAs`](https://github.com/quintel/etmoses/blob/master/app/services/testing_ground/save_as.rb))?